src/lib/vkimagesdatabase.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jolla.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef VKIMAGESDATABASE_H
+#define VKIMAGESDATABASE_H
+
+#include "abstractsocialcachedatabase_p.h"
+#include <QtCore/QDateTime>
+#include <QtCore/QStringList>
+
+struct VKUser
+{
+    VKUser(const QString &id, const QString &first_name, const QString &last_name,
+           const QString &photo_src, const QString &photo_file, int accountId);
+    VKUser() : accountId(0), photos_count(0) {}
+    QString id; // user id
+    QString first_name;
+    QString last_name;
+    QString photo_src;
+    QString photo_file;
+    int accountId;
+
+    int photos_count; // transient, not database-stored value! Not used in comparisons etc.
+
+    bool operator==(const VKUser &other) const;
+    bool operator!=(const VKUser &other) const { return !(*this == other); }
+};
+
+struct VKAlbum
+{
+    VKAlbum(const QString &id, const QString &owner_id, const QString &title,
+            const QString &description, const QString &thumb_src,
+            const QString &thumb_file, int size, int created, int updated,
+            int accountId);
+    VKAlbum() : size(0), created(0), updated(0), accountId(0) {}
+    QString id;       // album id
+    QString owner_id; // user id
+    QString title;
+    QString description;
+    QString thumb_src;
+    QString thumb_file;
+    int size;
+    int created;
+    int updated;
+    int accountId;
+
+    bool operator==(const VKAlbum &other) const;
+    bool operator!=(const VKAlbum &other) const { return !(*this == other); }
+};
+
+struct VKImage
```

Comment:
Also, I probably made a mistake in not putting these into a namespace, perhaps.  I wonder if they should be public classes within VKImageDatabase, instead of top level classes in the global namespace...